dbt/adapters/athena/lf_permissions.py

```diff
@@ -0,0 +1,149 @@
+from typing import Dict, List
+
+from mypy_boto3_lakeformation import LakeFormationClient
+from mypy_boto3_lakeformation.type_defs import (
+    BatchPermissionsRequestEntryTypeDef,
+    DataCellsFilterTypeDef,
+)
+from pydantic import BaseModel
+
+from dbt.adapters.athena.relation import AthenaRelation
+from dbt.events import AdapterLogger
+
+logger = AdapterLogger("AthenaLakeFormation")
+
+
+class FilterConfig(BaseModel):
+    row_filter: str
+    column_names: List[str] = []
+    principals: List[str] = []
+
+    def to_api_repr(self, catalog_id: str, database: str, table: str, name: str) -> DataCellsFilterTypeDef:
+        return {
+            "TableCatalogId": catalog_id,
+            "DatabaseName": database,
+            "TableName": table,
+            "Name": name,
+            "RowFilter": {"FilterExpression": self.row_filter},
+            "ColumnNames": self.column_names,
+            "ColumnWildcard": {"ExcludedColumnNames": []},
+        }
+
+    def to_update(self, existing: DataCellsFilterTypeDef) -> bool:
+        return self.row_filter != existing["RowFilter"]["FilterExpression"] or set(self.column_names) != set(
+            existing["ColumnNames"]
+        )
+
+
+class DataCellFiltersConfig(BaseModel):
+    enabled: bool = False
+    filters: Dict[str, FilterConfig]
+
+
+class LfGrantsConfig(BaseModel):
+    data_cell_filters: DataCellFiltersConfig
+
+
+class LfPermissions:
+    def __init__(self, catalog_id: str, relation: AthenaRelation, lf: LakeFormationClient) -> None:
+        self.catalog_id = catalog_id
+        self.relation = relation
+        self.database: str = relation.schema
+        self.table: str = relation.identifier
+        self.lf = lf
+
+    def get_filters(self) -> Dict[str, DataCellsFilterTypeDef]:
+        table_resource = {"CatalogId": self.catalog_id, "DatabaseName": self.database, "Name": self.table}
+        return {f["Name"]: f for f in self.lf.list_data_cells_filter(Table=table_resource)["DataCellsFilters"]}
+
+    def process_filters(self, config: LfGrantsConfig) -> None:
+        current_filters = self.get_filters()
+        logger.debug(f"CURRENT FILTERS: {current_filters}")
+
+        to_drop = [f for name, f in current_filters.items() if name not in config.data_cell_filters.filters]
```

Comment:
just a thought: how about edge cases where filters are renamed? it might be worth it to eventually compare the filter condition itself not just the name.
Imagine cases where somehow a filter is renamed in the lf_grant, or cases where pre-existing filters are exiting, and they need to be recreated via dbt-athena.

Reply:
if the pre-existing filter does not exist in config, it will be dropped and this seems to me OK because you should have 1 place for filters management: it may be dbt or any other tool if you managed to configure it. However if it's dbt, then config and actual state must be consistent

if existing filters are same as configs, then nothing will happen, dbt will just skip them

Reply:
ok let's don't complicate and I agree with you, filter management should be centralize in one place, therefore if the users have existing filters, they should be dropped (manually).

Let's focus on what can happen in the adapter: is it possible given the current implementation to rename a cell filter? 

If not, then we should compare filters by condition and name, drop them, and then recreate them WDTY?

Reply:
currently, the following logic is implemented:
1. If there is a filter with name, which is not specified in config, it will be dropped
2. If there is a filter with the same name as in config, then we check 2 conditions to make decision about updating: filter_expression and column_names must be same. Otherwise, it will be updated.
3. If there are no filters with name from config, just new filter will be created.

So that, if we need to rename existing filter, then current logic will drop existing one and create the new filter.

Reply:
ok great, then fine. Maybe you can cover that via a unit test?

Reply:
@nicor88 I'm not sure if it's possible now because `moto` does not support filter management according to [link to moto repo](https://github.com/getmoto/moto/blob/760fa8fdc0a6812a92e426e77ebef3bd3a27283c/moto/lakeformation/models.py#L141)

however you may try to do this using example model from this PR description

Reply:
Regarding moto support might be worth to add an issue to have it cover it in the future.